### PR TITLE
feature/9164-SubmitEvidence

### PR DESCRIPTION
**Pull request by Sparowhawk**
## Description of Change
Added submit evidence button to claims not in step 3 and put the workflow behind the feature flag submitEvidenceExpansion. Updated unit tests and e2e tests. Added submit evidence screen, updated workflow for file request and submit evidence to be full screen subtasks with cancels instead of back buttons and updated the copy for submit evidence scenario and allowed uploading files without an associated request

Had to update podfile as well

## Screenshots/Video
Submit Evidence workflow
Screenshot 2024-08-15 at 3 37 48 PM
Screenshot 2024-08-15 at 3 37 51 PM
Screenshot 2024-08-15 at 3 37 55 PM
Screenshot 2024-08-15 at 3 38 03 PM
Screenshot 2024-08-15 at 3 38 12 PM

## Testing
yarn test/detox also tested on Judy

- [x] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations
All
 Launch as a Fullscreen Subtask
 On completion/cancel, route user back to previous Child screen. So, either [Claim detail: Submit evidence] or [Claim detail: Review file request]
Submit evidence flow
 Create first screen; follow [copy deck](https://docs.google.com/spreadsheets/d/16waC665d7uNRQmM83SJIm9bTDyRePOuB6r9G_aQE8Tw/edit?gid=737285959#gid=737285959) and [design](https://www.figma.com/design/kSlXLwTVLunM0fqCG95k5x/Claims-2.0---%F0%9F%94%8D-Working---VA-Mobile?node-id=1987-1337&t=Zrxg1HCE8eXnE8qp-4)
 Make any copy updates to subsequent screens within the flow
Review File Request flow
 Make any copy updates to subsequent screens within the flow

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

File: VAMobile/src/api/claimsAndAppeals/uploadFileToClaim.tsx [hunk 1]
```diff
@@ -32,8 +32,8 @@ const uploadFileToClaim = ({ claimID, request, files }: UploadFileToClaimParamat
     const payload = JSON.parse(
       JSON.stringify({
         files: fileStrings,
-        trackedItemId: request.trackedItemId,
-        document_type: request.documentType,
+        trackedItemId: request?.trackedItemId,
```
Comment (dumathane) [inline VAMobile/src/api/claimsAndAppeals/uploadFileToClaim.tsx:35]: how do we confirm that trackedItemID can be nil on the backend? Did they make this change and is it associated to a ticket?

File: VAMobile/src/api/claimsAndAppeals/uploadFileToClaim.tsx [hunk 2]
```diff
@@ -82,22 +86,34 @@ const uploadFileToClaim = ({ claimID, request, files }: UploadFileToClaimParamat
...
+    claimData.attributes.eventsTimeline[indexOfRequest].documents = createFileRequestDocumentsArray(
+      files || [],
+      request?.trackedItemId || undefined,
+      request?.documentType || '',
+      dateUploadedString,
+    )
+    claimData.attributes.eventsTimeline[indexOfRequest].uploadDate = dateUploadedString
+  } else {
+    const claimEvent: ClaimEventData = {
+      type: '',
```
Comment (dumathane) [inline VAMobile/src/api/claimsAndAppeals/uploadFileToClaim.tsx:108]: same here. We can just nil a type?

File: VAMobile/src/api/claimsAndAppeals/uploadFileToClaim.tsx [hunk 3]
```diff
@@ -82,22 +86,34 @@ const uploadFileToClaim = ({ claimID, request, files }: UploadFileToClaimParamat
...
+      date: dateUploadedString,
+      status: FILE_REQUEST_STATUS.SUBMITTED_AWAITING_REVIEW,
+      uploaded: true,
+      uploadDate: dateUploadedString,
+      documents: createFileRequestDocumentsArray(files || [], undefined, '', dateUploadedString),
```
Comment (dumathane) [inline VAMobile/src/api/claimsAndAppeals/uploadFileToClaim.tsx:113]: blank type again.

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.tsx [hunk 4]
```diff
@@ -78,38 +79,59 @@ function UploadFile({ navigation, route }: UploadFileProps) {
...
+            request?.type || 'Submit Evidence',
+            'file',
+          ),
+        )
         showSnackBar(snackbarMessages.successMsg, dispatch, undefined, true)
       },
       onError: () => showSnackBar(snackbarMessages.errorMsg, dispatch, onUploadConfirmed, false, true),
     }
-    const params: UploadFileToClaimParamaters = { claimID, request, files: filesList }
+    const params: UploadFileToClaimParamaters = { claimID, documentType: documentType, request, files: filesList }
```
Comment (alexandec) [inline VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.tsx:134]: Can use shorthand for documentType, same as other params

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimDetailsScreen.tsx [hunk 5]
```diff
@@ -210,6 +219,17 @@ function ClaimDetailsScreen({ navigation, route }: ClaimDetailsScreenProps) {
...
+        const buttonProps: ButtonProps = {
+          buttonType: ButtonVariants.Primary,
+          label: t('claimDetails.submitEvidence'),
+          onPress: submitEvidencePress,
+        }
+        return (
+          <Box mt={theme.dimensions.standardMarginBetween}>
+            <Button {...buttonProps} />
+          </Box>
+        )
```
Comment (dumathane) [inline VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimDetailsScreen.tsx:232]: image
design has gutters.

**Review by alexandec — CHANGES_REQUESTED** [review]

**Comment by alexandec** [comment]
Also noticed a conflict on the Claims E2E tests

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.tsx [hunk 6]
```diff
@@ -78,38 +79,59 @@ function UploadFile({ navigation, route }: UploadFileProps) {
 
   useEffect(() => {
     if (filesUploadedSuccess) {
-      navigateTo('FileRequest', { claimID: claim?.id || '' })
+      navigateTo('ClaimDetailsScreen', { claimID: claimID, claimType: ClaimTypeConstants.ACTIVE })
     }
-  }, [filesUploadedSuccess, claim, navigateTo])
+  }, [filesUploadedSuccess, claimID, navigateTo])
```
Comment (dumathane) [inline VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.tsx:84]: scope creep? Why is this being changed? and shouldn't claimType be in the dependency array?

Comment (dumathane) [reply VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.tsx:84]: okay i get the claimID part.

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.tsx [hunk 7]
```diff
@@ -78,38 +79,59 @@ function UploadFile({ navigation, route }: UploadFileProps) {
...
-  }, [documentType])
+      })
+    }
+  }, [documentType, originalRequest])
 
   const onUploadConfirmed = () => {
-    logAnalyticsEvent(Events.vama_evidence_cont_3(claim?.id || '', request.trackedItemId || null, request.type, 'file'))
+    logAnalyticsEvent(
+      Events.vama_evidence_cont_3(
+        claim?.id || '',
```
Comment (dumathane) [inline VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.tsx:102]: why is this not claimID ?

File: VAMobile/src/utils/waygateConfig.ts
Comment (dumathane) [inline VAMobile/src/utils/waygateConfig.ts:1]: I don't think we actually need to do this, but it's fine to have.

**Review by dumathane — DISMISSED** [review]
Need to fix submit evidence button.

File: VAMobile/src/api/claimsAndAppeals/uploadFileToClaim.tsx (hunk 2 above)
Comment (Sparowhawk) [reply VAMobile/src/api/claimsAndAppeals/uploadFileToClaim.tsx:108]: for this one it can be nil since this is only adding the file locally so that we can add it to the files list

File: VAMobile/src/api/claimsAndAppeals/uploadFileToClaim.tsx (hunk 3 above)
Comment (Sparowhawk) [reply VAMobile/src/api/claimsAndAppeals/uploadFileToClaim.tsx:113]: same as above this is for us locally so we can display the file in the list so it can be nil

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimDetailsScreen.tsx (hunk 5 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimDetailsScreen.tsx:232]: fixed

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SubmitEvidence.tsx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  const onPhotoPress = () => {
+    logAnalyticsEvent(Events.vama_evidence_start(claimID, null, 'Submit Evidence', 'photo'))
+    navigateTo('TakePhotos', { claimID })
+  }
+
+  return (
+    <FullScreenSubtask
+      leftButtonText={t('close')}
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SubmitEvidence.tsx:35]: changed to cancel

File: VAMobile/src/utils/waygateConfig.ts
Comment (Sparowhawk) [reply VAMobile/src/utils/waygateConfig.ts:1]: leaving this in just in case

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.tsx (hunk 7 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.tsx:102]: updated all of them to claimID

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.tsx (hunk 6 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.tsx:83]: ClaimType doesn't need to be. Since they can only upload on an active non closed claim we can use the constant for active, which doesn't need to be included

File: VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.tsx (hunk 4 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/ClaimsScreen/ClaimDetailsScreen/ClaimStatus/ClaimFileUpload/SelectFile/UploadFile/UploadFile.tsx:128]: updated

File: VAMobile/src/api/claimsAndAppeals/uploadFileToClaim.tsx [hunk 9]
```diff
@@ -23,7 +23,7 @@ import { claimsAndAppealsKeys } from './queryKeys'
 /**
  * Uploads a File to Claim
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/api/claimsAndAppeals/uploadFileToClaim.tsx:24]: added

File: VAMobile/src/api/claimsAndAppeals/uploadFileToClaim.tsx [hunk 10]
```diff
@@ -82,22 +86,34 @@ const uploadFileToClaim = ({ claimID, request, files }: UploadFileToClaimParamat
  */
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/api/claimsAndAppeals/uploadFileToClaim.tsx:86]: added

File: VAMobile/e2e/tests/Claims.e2e.ts [hunk 11]
```diff
@@ -301,10 +228,87 @@ describe(':ios: Claims Screen', () => {
...
+  it('verify the select files screen displays the correct info', async () => {
+    await element(by.text(ClaimsE2eIdConstants.SELECT_A_FILE_TEXT)).tap()
+    await expect(element(by.text('Select a file to upload'))).toExist()
+    await expect(element(by.label(ClaimsE2eIdConstants.MAXIMUM_FILE_SIZE_LABEL))).toExist()
+    await expect(element(by.text(ClaimsE2eIdConstants.ACCEPTED_FILE_TYPES_TEXT))).toExist()
+    await expect(element(by.id(ClaimsE2eIdConstants.SELECT_A_FILE_TEXT)).atIndex(1)).toExist()
+  })
+
+  it('verify tap select a file options', async () => {
+    if (device.getPlatform() === 'android') {
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/e2e/tests/Claims.e2e.ts:262]: all uses of this just changes the way to find an element vs iOS. The whole test isn't for just android, there is an individual thing that needs to be found this way and then the rest of the test is the same for iOS and Android.

File: VAMobile/e2e/tests/Claims.e2e.ts [hunk 12]
```diff
@@ -301,10 +228,87 @@ describe(':ios: Claims Screen', () => {
         by.text('Your claim has been assigned to a reviewer who is determining if additional information is needed.'),
       ),
     ).not.toExist()
-    await element(by.text(CommonE2eIdConstants.CLAIMS_HISTORY_BUTTON_TEXT)).tap()
+  })
+
+  it('should verify submit evidence button exists', async () => {
+    await element(by.id(ClaimsE2eIdConstants.CLAIMS_DETAILS_SCREEN_ID)).scrollTo('top')
+    await expect(element(by.text('Submit evidence'))).toExist()
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/e2e/tests/Claims.e2e.ts]: updated

File: VAMobile/src/api/claimsAndAppeals/uploadFileToClaim.tsx [hunk 13]
```diff
@@ -67,8 +67,12 @@ const uploadFileToClaim = ({ claimID, request, files }: UploadFileToClaimParamat
       ),
     )
 
-    formData.append('trackedItemId', JSON.parse(JSON.stringify(request.trackedItemId)))
-    formData.append('documentType', JSON.parse(JSON.stringify(request.documentType)))
+    if (request) {
+      formData.append('trackedItemId', JSON.parse(JSON.stringify(request.trackedItemId)))
+      formData.append('documentType', JSON.parse(JSON.stringify(request.documentType)))
+    } else {
+      formData.append('documentType', JSON.parse(JSON.stringify(documentType)))
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/api/claimsAndAppeals/uploadFileToClaim.tsx]: updated for documentType, the request ones have to remain that ways since they technically can be null or undefined

File: VAMobile/src/api/claimsAndAppeals/uploadFileToClaim.tsx [hunk 14]
```diff
@@ -32,8 +32,8 @@ const uploadFileToClaim = ({ claimID, request, files }: UploadFileToClaimParamat
     const payload = JSON.parse(
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/api/claimsAndAppeals/uploadFileToClaim.tsx:35]: this was here from the slice migration and I didn't want to remove it. I also see that it might be needed since technically request.trackedItemID can resolve to null or undefined

**Review by alexandec — DISMISSED** [review]

**Review by dumathane — DISMISSED** [review]

**Review by dumathane — APPROVED** [review]

**Review by rbontrager — APPROVED** [review]

**Review by alexandec — APPROVED** [review]
